binder/plugins/views/csvexport.py

```diff
@@ -1,16 +1,172 @@
-from binder.router import list_route
-from abc import ABCMeta
+import abc
 import csv
+from tempfile import NamedTemporaryFile
+from typing import List
+
+from django.http import HttpResponse, HttpRequest
+
 from binder.json import jsonloads
-from django.http import HttpResponse
+from binder.router import list_route
+
+
+class ExportFileAdapter:
+	"""
+	Adapter between the data that is exported, and the export file
+	"""
+	__metaclass__ = abc.ABCMeta
+
+	def __init__(self, request: HttpRequest, csv_settings: 'CsvExportView.CsvExportSettings'):
+		self.csv_settings = csv_settings
+		self.request = request
+
+	@abc.abstractmethod
+	def set_file_name(self, file_name: str):
+		"""
+		Sets the file name of the file that needs to be export. File name does not have the extension.
+
+		e.g. set_file_name('foo') => file download name is 'foo.csv' or 'foo.xlsx'
+
+		:param file_name:
+		:return:
+		"""
+		pass
+
+	@abc.abstractmethod
+	def set_columns(self, columns: List[str]):
+		"""
+		Set the column names of the file
+
+		:param columns:
+		:return:
+		"""
+		pass
+
+
+	@abc.abstractmethod
+	def add_row(self, values: List[str]):
+		"""
+		Add a row with values to the file
+
+		:param values:
+		:return:
+		"""
+		pass
+
+
+
+	@abc.abstractmethod
+	def get_response(self) -> HttpResponse:
+		"""
+		Return a http response with the content of the file
+
+		:param columns:
+		:return:
+		"""
+		pass
+
+
+class CsvFileAdapter(ExportFileAdapter):
+	"""
+	Adapter for returning CSV files
+	"""
+
+	def __init__(self, request: HttpRequest, csv_settings: 'CsvExportView.CsvExportSettings'):
+		super().__init__(request, csv_settings)
+		self.response = HttpResponse(content_type='text/csv')
+		self.file_name = 'export'
+		self.writer = csv.writer(self.response)
+
+	def set_file_name(self, file_name: str):
+		self.file_name = file_name
+
+	def set_columns(self, columns: List[str]):
+		self.writer.writerow(list(map(lambda x: x[1], self.csv_settings.column_map)))
```

Comment:
Shouldnt this just write the provided columns as a row? That would make it more consistent with the excel adapter. Also this is the only place where an adapter uses the csv settings, if this is changed to be the columns maybe the adapter doesnt need the csv settings.